packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -306,6 +306,8 @@ Dialog.slotClassNames = {
 
 /**
  * A Dialog displays important information on top of a page which usually requires user's attention, confirmation or interaction.
+ * Dialogs are purposefully interruptive, so they should be used sparingly.
+ *
```

Comment:
Would be great to have some text about how to create modal / non-modal dialogs, or at least hint about using focusTrap...